pkg/flags/github.go

```diff
@@ -56,4 +70,87 @@ func (g *GitHubFlags) Register(set *cli.FlagSet) {
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-app-private-key-resource-name",
+		EnvVar: "GITHUB_APP_PRIVATE_KEY_RESOURCE_NAME",
+		Target: &g.FlagGitHubAppPrivateKeyResourceName,
+		Usage:  "The resource name of the private key to use with the GitHub App.",
+	})
+
+	set.AfterParse(func(merr error) error {
+		if g.FlagGitHubToken == "" && g.FlagGitHubAppID == "" {
+			merr = errors.Join(merr, fmt.Errorf("one of github token or github app id are required"))
+		}
+		if g.FlagGitHubToken != "" && g.FlagGitHubAppID != "" {
+			merr = errors.Join(merr, fmt.Errorf("only one of github token or github app id are allowed"))
+		}
+		if g.FlagGitHubAppID != "" && g.FlagGitHubAppInstallationID == "" {
+			merr = errors.Join(merr, fmt.Errorf("a github app installation id is required when using a github app id"))
+		}
+		if g.FlagGitHubAppID != "" && g.FlagGitHubAppPrivateKeyResourceName == "" {
+			merr = errors.Join(merr, fmt.Errorf("a github app installation id is required when using a github app id"))
+		}
+		return merr
+	})
+}
+
+func (g *GitHubFlags) GetTokenSource(ctx context.Context, permissions map[string]string) (githubauth.TokenSource, error) {
+	if g.FlagGitHubToken != "" {
+		githubTokenSource, err := githubauth.NewStaticTokenSource(g.FlagGitHubToken)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create github static token source: %w", err)
+		}
+		return githubTokenSource, nil
+	} else {
+		sm, err := secretmanager.NewClient(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create secret manager client: %w", err)
+		}
+		defer sm.Close()
+
+		privateKeyPEM, err := accessSecret(ctx, sm, g.FlagGitHubAppPrivateKeyResourceName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get github private key pem: %w", err)
+		}
+
+		app, err := githubauth.NewApp(g.FlagGitHubAppID, g.FlagGitHubAppInstallationID, privateKeyPEM)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create github app token source: %w", err)
+		}
+
+		return app.SelectedReposTokenSource(permissions, g.FlagGitHubRepo), nil
+	}
+}
+
+// AccessSecret reads a secret from Secret Manager using the given client and
+// validates that it was not corrupted during retrieval. The secretResourceName
+// should be in the format: 'projects/*/secrets/*/versions/*'.
+func accessSecret(ctx context.Context, client *secretmanager.Client, secretResourceName string) (string, error) {
+	result, err := client.AccessSecretVersion(ctx, &secretmanagerpb.AccessSecretVersionRequest{
+		Name: secretResourceName,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to access secret %s: %w", secretResourceName, err)
+	}
+	crc32c := crc32.MakeTable(crc32.Castagnoli)
+	checksum := int64(crc32.Checksum(result.Payload.Data, crc32c))
+	if checksum != *result.Payload.DataCrc32C {
+		return "", fmt.Errorf("failed to access secret %s: data corrupted", secretResourceName)
+	}
+	return string(result.Payload.Data), nil
```

Comment:
I'd prefer if we decouple from secrets management in the app code. Can we use Cloud Run's native Secret Manager integration instead to source this from the environment? That will make our service more portable.